### PR TITLE
Add Nightly tests docs in the CONTRIBUTING page

### PR DESCRIPTION
**Pull request by mariusc83**
### What does this PR do?

A brief description of the change being made with this pull request.

### Motivation

What inspired you to submit this pull request?

### Additional Notes

Anything else we should know when reviewing?

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [ ] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -276,5 +276,39 @@ Because we sometimes need to reuse some setup or assertions in our tests, we ten
...
 - `fun setupSomething()`: method to setup a complex test (should only be used in the Given part of a test).
 
+### Nightly Tests
 
+Each public API method in the SDK is covered by a test case in the `nightly-tests` module. All test cases are executed on a Bitrise emulator by a Datadog Synthetic Test every 24 hours. Each test case
+output is measured by 2 Datadog Monitors (one for performance and one for functionality). There are some best practices when writing a nightly test as follows: 
+
+- The test method name must follow the following format: [rootFeature]_[subFeature]_[method]_[additionalInfo] where:
+
+    1. rootFeature is one of the top level features (logs, rum, apm, et)
```
Comment (0xnm) [inline CONTRIBUTING.md:286]: probably it is better to apply code span to the `rootFeature` name and also to the cases below.

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -276,5 +276,39 @@ Because we sometimes need to reuse some setup or assertions in our tests, we ten
...
+    2. subFeature is the feature under test (logger, monitor, …)
+    
+    3. method is the method under test (not necessarily exactly the exact method name but the purpose of the feature from a customer’s PoV, e.g: DataScrubbing)
+    
+    4. additionalInfo is some context to distinguish multiple test on the same method (could be related to the argument, the context, a state)
+    
+- We need to add an identifier in the method documentation following the method signature in the [apiSurface](dd-sdk-android/apiSurface). 
+  This will be used by our test coverage tool.
+  
+- Each tested Public API call has to be scoped inside a Span which will be used in the performance monitor
```
Comment (0xnm) [inline CONTRIBUTING.md:297]:
```suggestion
- Each tested Public API call has to be scoped inside a Span which will be used in the performance monitor.
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -276,5 +276,39 @@ Because we sometimes need to reuse some setup or assertions in our tests, we ten
...
+
+
+We have created a Live Template that you can add in your development environment (AndroidStudion, IntelliJ) to ease your work when creating a nightly test:
```
Comment (0xnm) [inline CONTRIBUTING.md:300]:
```suggestion
We have created a Live Template that you can add in your development environment (Android Studio, IntelliJ IDEA) to ease your work when creating a nightly test:
```

**Review by 0xnm — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -276,5 +276,38 @@ Because we sometimes need to reuse some setup or assertions in our tests, we ten
...
+    2. `subFeature` is the feature under test (logger, monitor, …)
+    
+    3. `method` is the method under test (not necessarily exactly the exact method name but the purpose of the feature from a customer’s PoV, e.g: DataScrubbing)
+    
+    4. `additionalInfo` is some context to distinguish multiple test on the same method (could be related to the argument, the context, a state)
+    
+- We need to add an identifier in the method documentation following the method signature in the [apiSurface](dd-sdk-android/apiSurface). 
+  This will be used by our test coverage tool.
+  
+- Each tested Public API call has to be scoped inside a Span which will be used in the performance monitor.
```
Comment (xgouchet) [inline CONTRIBUTING.md:297]: This part doesn't necessarily need to be public, since third party contributors can't create monitors in our testing org

**Review by xgouchet — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 4 above)
Comment (mariusc83) [reply CONTRIBUTING.md:297]: Which one ? the last point about the `performance monitor` or the whole best practices ? I would rather keep the first 2 points.

Comment (xgouchet) [reply CONTRIBUTING.md:297]: No just the last line about creating a performance monitor

**Review by xgouchet — APPROVED** [review]

**Pull request merged**